File: Java/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int numer3= numer1 * denom2 + numer2 * denom1;
        int denom3= denom1 * denom2;
        for(int i=Math.min(numer3, denom3); i >=2 ; i-- ){
            if (numer3 % i ==0 && denom3 % i ==0) {
                numer3 /= i;
                denom3 /= i;
            }
        }
        int[] answer = {numer3, denom3};
        return answer;
    }
}

File: Python/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    numer3= (numer1 * denom2)+ (numer2 * denom1)
    denom3= denom1 * denom2
    # min 함수 내장되어있음
    for i in range(min(numer3, denom3),0,-1):
        if (numer3%i==0 and denom3%i==0):
            numer3 //= i
            denom3 //= i
    answer = [numer3, denom3]
    return answer